test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/OpenIdConnectHandlerTests.cs

```diff
@@ -443,12 +275,12 @@ private static TestServer CreateServer(IOptions<OpenIdConnectAuthenticationOptio
             );
         }
 
-        private static TestServer CreateServer(CustomConfigureOptions configureOptions, ILoggerFactory loggerFactory)
+        private static TestServer CreateServer(CustomConfigureOptions configureOptions, IUrlEncoder encoder, ILoggerFactory loggerFactory, OpenIdConnectAuthenticationHandler handler = null)
```

Comment:
Instead of CustomConfigureOptions, this should really take either `Action<OpenIdOptions>`, or if you want full control over all services to be able to configure everything `Action<IServiceCollection>`.  All of this test code should be setting up the services via DI and replacing them with the appropriate test services, not injecting them via some new parallel custom test code path that is different from the normal code path...